File: library/src/main/java/ke/tang/refresh/ViewRefreshable.java
package ke.tang.refresh;

import android.view.View;

class ViewRefreshable implements RefreshLayout.Refreshable {
    private View mTarget;

    public ViewRefreshable(View target) {
        mTarget = target;
    }

    @Override
    public boolean isIndicator() {
        return false;
    }

    @Override
    public void onOffset(float delta) {

    }

    @Override
    public void onRelease(boolean isTrigger) {

    }

    @Override
    public void onReset() {

    }

    @Override
    public int getContentSize() {
        return mTarget.getHeight();
    }
}

File: library/src/main/java/ke/tang/refresh/ViewRefreshable.kt
package ke.tang.refresh

import android.view.View
import android.view.ViewGroup

internal class ViewRefreshable(private val mView: View) : Refreshable {
    override fun isIndicator() = false

    override fun onOffset(delta: Float) {
    }

    override fun onRelease(isTrigger: Boolean) {
    }

    override fun onReset() {
    }

    override fun getContentSize(isRefreshVertical: Boolean) = if (isRefreshVertical) {
        mView.height + with(mView.layoutParams as ViewGroup.MarginLayoutParams) { topMargin + bottomMargin }
    } else {
        mView.width + with(mView.layoutParams as ViewGroup.MarginLayoutParams) { leftMargin + rightMargin }
    }
}